commit dc02e73330344fcdf8fdea71c65b884372fdd3c2 (refs/changes/32/3232/1)
Author: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
Date:   2018-01-18T09:07:09-07:00 (1 year, 9 months ago)
    
    MANTA-3417 long-running query probe does not work with PostgreSQL 9.6.3

diff --git a/alarm_metadata/probe_templates/postgres.yaml b/alarm_metadata/probe_templates/postgres.yaml
index 323e5f3..b9f97fa 100644
--- a/alarm_metadata/probe_templates/postgres.yaml
+++ b/alarm_metadata/probe_templates/postgres.yaml
@@ -243,7 +243,9 @@
         -
             type: cmd
             config:
-                cmd: "PATH=/opt/postgresql/current/bin:/opt/local/bin:$PATH psql -U postgres -c \"SELECT pid,usename,client_addr,client_port,waiting,state,NOW() - query_start AS elapsed, query FROM pg_stat_activity WHERE state != 'idle' AND query NOT LIKE 'autovacuum: %' AND NOW() - query_start > interval '120 seconds'\" | tail +3"
+                cmd: "if [ $(psql -V | cut -f 3 -d ' ' | cut -f 1,2 -d '.') = '9.6' ]; then /opt/postgresql/current/bin/psql -U postgres -c \"SELECT pid,usename,client_addr,client_port,wait_event,wait_event_type,state,NOW() - query_start AS elapsed, query FROM pg_stat_activity WHERE state != 'idle' AND query NOT LIKE 'autovacuum: %' AND NOW() - query_start > interval '120 seconds'\" | tail +3; else /opt/postgresql/current/bin/psql -U postgres -c \"SELECT pid,usename,client_addr,client_port,waiting,state,NOW() - query_start AS elapsed, query FROM pg_stat_activity WHERE state != 'idle' AND query NOT LIKE 'autovacuum: %' AND NOW() - query_start > interval '120 seconds'\" | tail +3; fi"
+                env:
+                  PATH: "/opt/postgresql/current/bin:/opt/local/bin:$PATH"
                 interval: 60
                 timeout: 30
                 threshold: 2
